Framework/Utils/test/test_RootTreeWriterWorkflow.cxx

```diff
@@ -22,32 +22,49 @@
 #include "Headers/DataHeader.h"
 #include "../../Core/test/TestClasses.h"
 #include "FairMQLogger.h"
-#include <TSystem.h>
 #include <TTree.h>
 #include <TFile.h>
 #include <vector>
+#include <stdexcept>
+// note: std filesystem is first supported in gcc 8
+#include <boost/filesystem.hpp>
 
 using namespace o2::framework;
 
-#define ASSERT_ERROR(condition)                                   \
-  if ((condition) == false) {                                     \
-    LOG(ERROR) << R"(Test condition ")" #condition R"(" failed)"; \
-  }
-
+// a helper class to do the checking at the end of the program when
+// the destructor of the class is called.
 class StaticChecker
 {
  public:
   StaticChecker() = default;
-  ~StaticChecker()
+  // have to throw an exception in the destructor if checking fails
+  // this is ok in this cae because no other instances which would require proper
+  // cleanup are expected
+  ~StaticChecker() noexcept(false)
   {
     for (auto const& check : mChecks) {
       TFile* file = TFile::Open(check.first.c_str());
-      assert(file != nullptr);
-      TTree* tree = file != nullptr ? reinterpret_cast<TTree*>(file->GetObjectChecked("testtree", "TTree")) : nullptr;
-      assert(tree != nullptr);
+      if (file == nullptr) {
+        setError(std::string("missing file ") + check.first.c_str());
+        continue;
+      }
+      TTree* tree = reinterpret_cast<TTree*>(file->GetObjectChecked("testtree", "TTree"));
       if (tree) {
-        assert(tree->GetEntries() == check.second);
+        if (tree->GetEntries() != check.second) {
+          setError(std::string("inconsistent number of entries in 'testtree' of file ") + check.first.c_str() + " expecting " + std::to_string(check.second) + " got " + std::to_string(tree->GetEntries()));
+        }
+      } else {
+        setError(std::string("can not find tree 'testtree' in file ") + check.first.c_str());
       }
+      file->Close();
+      // TODO: keeping this as a note because this is not understood
+      // files should get deleted at this point, but if this line is in, the files are already deleted when entering
+      // the loop. Not clear if there is some out-of-order execution
+      // could not test with std filesystem, because the is first supported in gcc 8
+      //boost::filesystem::remove(check.first.c_str());
```

Comment:
for the record: this is understood. All processes have the same checker setup, the "winner takes it all" and deletes the files, the other processes have nothing to check. So this is a serious flaw in the unit test. The `StaticChecker` approach is not appropriate. Fix is around the corner.